src/backend/distributed/commands/statistics.c

```diff
@@ -354,14 +356,30 @@ PostprocessAlterStatisticsSchemaStmt(Node *node, const char *queryString)
  * was set to true.
  */
 List *
-AlterStatisticsSchemaStmtObjectAddress(Node *node, bool missingOk)
+AlterStatisticsSchemaStmtObjectAddress(Node *node, bool missingOk, bool isPostprocess)
 {
 	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
 
 	ObjectAddress *address = palloc0(sizeof(ObjectAddress));
-	String *statName = llast((List *) stmt->object);
-	Oid statsOid = get_statistics_object_oid(list_make2(makeString(stmt->newschema),
-														statName), missingOk);
+	Oid statsOid = InvalidOid;
+
+	List *statName = (List *) stmt->object;
+
+	if (isPostprocess)
+	{
+		/*
```

Comment:
lets add test for this one? The test that Marco suggested in the earlier PR

Reply:
There is a statement 'ALTER STATISTICS s7 SET SCHEMA test_alter_schema;' in propagate_statistics.sql to call address method with isPostprocess set to true. 
We also call the address method with isPostProcess set to false during pg vanilla test in alter_generic.sql.